OpenRA.Mods.Common/AI/HackyAI.cs

```diff
@@ -656,8 +619,7 @@ void AssignRolesToIdleUnits(Actor self)
 			if (--assignRolesTicks <= 0)
 			{
 				assignRolesTicks = Info.AssignRolesInterval;
-				if (resLayer != null && !resLayer.IsResourceLayerEmpty)
-					GiveOrdersToIdleHarvesters();
+				harvManager.GiveOrdersToIdleHarvesters(activeUnits);
```

Comment:
Can we rename this to `AssignRoles` or `Tick` or something else a bit more generic?